NMCA_001/NMCA_001_script.lua

```diff
@@ -0,0 +1,767 @@
+
+    ScenarioInfo.UEFTech:SetCanTakeDamage(false)
+    ScenarioInfo.UEFTech:SetCanBeKilled(false)
+    ScenarioInfo.UEFTech:SetDoNotTarget(true)
+    ScenarioInfo.UEFTech:SetReclaimable(false)
+
+	local Walls = ScenarioUtils.CreateArmyGroup('UEF', 'M1Walls')
+
+	ScenarioInfo.Ship:SetCustomName('High Command Ship')
+	ScenarioInfo.UEFTech:SetCustomName('Tech Centre')
+	
+	for _, v in Engineers do
+        IssueMove({v}, ScenarioUtils.MarkerToPosition('M1_Nomads_Engineer_Move_Marker'))
+    end
+
+    local units = ScenarioUtils.CreateArmyGroupAsPlatoon('UEF', 'AirPatrol_' .. Difficulty, 'NoFormation')
+
+    for k, v in units:GetPlatoonUnits() do
+        ScenarioFramework.GroupPatrolRoute({v}, ScenarioPlatoonAI.GetRandomPatrolRoute(ScenarioUtils.ChainToPositions('M1_AirPatrol_Chain')))
+    end
+
+    local LabPatrol_1 = ScenarioUtils.CreateArmyGroupAsPlatoon('UEF', 'LabPatrol_1', 'GrowthFormation')
+
+    for k, v in LabPatrol_1:GetPlatoonUnits() do
+        ScenarioFramework.GroupPatrolRoute({v}, ScenarioPlatoonAI.GetRandomPatrolRoute(ScenarioUtils.ChainToPositions('LabPatrol_1_Chain')))
+    end
+
+    local LabPatrol_2 = ScenarioUtils.CreateArmyGroupAsPlatoon('UEF', 'LabPatrol_2', 'GrowthFormation')
+
+    for k, v in LabPatrol_2:GetPlatoonUnits() do
+        ScenarioFramework.GroupPatrolRoute({v}, ScenarioPlatoonAI.GetRandomPatrolRoute(ScenarioUtils.ChainToPositions('LabPatrol_2_Chain')))
+    end
+
+    local LabPatrol_3 = ScenarioUtils.CreateArmyGroupAsPlatoon('UEF', 'LabPatrol_3', 'GrowthFormation')
+
+    for k, v in LabPatrol_3:GetPlatoonUnits() do
+        ScenarioFramework.GroupPatrolRoute({v}, ScenarioPlatoonAI.GetRandomPatrolRoute(ScenarioUtils.ChainToPositions('LabPatrol_3_Chain')))
+    end
+
+    ----------
+	-- Run AI
+	----------
+    M1OutpostAI.UEFOutpostAI()
+
+    ----------
+	-- Cinematics
+	----------
+	ForkThread(M1NIS)
+end
+
+function PlayerWin()
+    KillGame()
+end
+
+function PlayerLose(Ship)
+	----------
+	-- Player Lost Game
+	----------
+    PlayerLost = true
+
+
+    Ship:SetCanBeKilled(false)
+    Ship:SetReclaimable(false)
+
+    -- Mark objectives as failed
+    for k, v in AssignedObjectives do
+        if(v and v.Active) then
+            v:ManualResult(false)
+        end
+    end
+
+	ScenarioFramework.CDRDeathNISCamera(Ship)
+
+	local UEFUnits = GetUnitsInRect(ScenarioUtils.AreaToRect('M1'))
+    M1OutpostAI.DisableBase()
+    ScenarioFramework.EndOperationSafety()
+	
+	for k, v in UEFUnits do
+	    if v and not v:IsDead() and (v:GetAIBrain() == ArmyBrains[UEF]) then
+            v:Stop()
+	    end
+	end
+
+    local NISUnits = ScenarioUtils.CreateArmyGroup('UEF', 'PlayerLose')
+    for k, v in NISUnits do
+    	v:SetFireState('HoldFire')
+        IssueAttack({v}, Ship)
+    end
+end
+
+function KillGame()
+    ForkThread(
+        function()
+            ScenarioFramework.EndOperation(ScenarioInfo.OpComplete, ScenarioInfo.OpComplete)

```

Comment:
this function accepts 3 arguments, 3rd is secondary objectives if they are complete or not